Automattic/jetpack
# Sync: Implement term taxonomy blacklist

Because of conflicts with filters, we'd like to stop syncing some taxonomies. To take care of that, this PR introduces a term taxonomy blacklist, similar to the one we have for post types. It also 

#### Changes proposed in this Pull Request:
* Define taxonomy blacklist with possible `get_*` filter conflicts
* Implement the taxonomy blacklist setting
* Filter out blacklisted taxonomies when syncing terms
* Exclude blacklisted taxonomies from term histogram
* Add taxonomy blacklist tests

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Related with p5TWut-hM-p2 and p7rcWF-12N-p2

#### Testing instructions:
* Checkout this branch.
* Register a forbidden custom taxonomy (for example `bloginfo_rss`).
* Create a term in that taxonomy, and verify it doesn't incrementally sync.
* Trigger full sync, and verify it doesn't sync forbidden taxonomy terms.
* Make sure tests pass.

#### Proposed changelog entry for your changes:
* Sync: Implement term taxonomy blacklist
